consensus/hotstuff/signature/randombeacon_reconstructor.go

```diff
@@ -1,52 +1,85 @@
 package signature
 
 import (
+	"fmt"
+
 	"github.com/onflow/flow-go/consensus/hotstuff"
 	"github.com/onflow/flow-go/crypto"
 	"github.com/onflow/flow-go/model/flow"
 )
 
 // RandomBeaconReconstructor implements hotstuff.RandomBeaconReconstructor.
-// The implementation wraps the thresholdSigner and translates the signer identity into signer index.
+// The implementation wraps the hotstuff.RandomBeaconInspector and translates the signer identity into signer index.
+// It has knowledge about DKG to be able to map signerID to signerIndex
 type RandomBeaconReconstructor struct {
 	dkg                   hotstuff.DKG                   // to lookup signer index by signer ID
 	randomBeaconInspector hotstuff.RandomBeaconInspector // a stateful object for this block. It's used for both storing all sig shares and producing the node's own share by signing the block
 }
 
 var _ hotstuff.RandomBeaconReconstructor = &RandomBeaconReconstructor{}
 
-func NewRandomBeaconReconstructur(dkg hotstuff.DKG, randomBeaconInspector hotstuff.RandomBeaconInspector) *RandomBeaconReconstructor {
+func NewRandomBeaconReconstructor(dkg hotstuff.DKG, randomBeaconInspector hotstuff.RandomBeaconInspector) *RandomBeaconReconstructor {
```

Comment:
I think `NewRandomBeaconReconstructor` may not need an `randomBeaconInspector` input. Using the `hotstuff.DKG` it can retrieve the data needed to construct `randomBeaconInspector` inside the function.

Reply:
Need a bit more info, but yeah, makes sense to include that.